src/responder/nss/nsssrv.c

```diff
@@ -540,6 +541,20 @@ int main(int argc, const char *argv[])
     /* set up things like debug, signals, daemonization, etc... */
     debug_log_file = "sssd_nss";
 
+    if (is_socket_activated()) {
+        char *logpath;
+        int fd;
+
+        ret = asprintf(&logpath, "%s/%s.log", LOG_PATH, debug_log_file);
+        if (ret == -1) {
+            return ENOMEM;
+        }
+
+        fd = open(logpath, O_RDONLY);
+        fchown(fd, 0, 0);
+        close(fd);
+    }
+
```

Comment:
Hmm. It *does* make a lot of sense, but also makes things a little bit more complicated.

The approach currently done (please, not consider socket-activation here) is:
- server_setup();
  - chown_debug_files();
  - become_user();
  ...
  - open_debug_file();

So, by your comment I'm considering the current behavior is also wrong and what I had in mind was a fix for both by opening and changing the owner of the file before dropping the privileges (considering I don't need the event_context created at this point, as it's created later in server_setup()).

In order to follow with my approach, I'd also have to set up log rotation before opening the file (and so far I guess it's okay ...). Now, there are two parts which made me a little bit confused ...

1) In case the debug_to_file is disabled in sssd.conf
    My understanding is that if that's the case I could simply close the recently opened file and that's it. Or am I mistaken?

2) I'd go for a completely new function instead of hacking open_debug_file_ex() and open_debug_file(). And what I have in mind is something towards these lines ...

```
int open_and_fchown_debug_file(uid_t uid, gid_t gid)
{
    FILE *f;
    char *logpath;
    int ret;

    ret = asprintf(&logpath, "%s/%s.log", LOG_PATH, debug_log_file);
    if (ret == -1) {
        ret = ENOMEM;
        goto done;
    }

    errno = 0;
    fd = open(logpath, FD_CLOEXEC, O_WRONLY);
    if (fd == -1) {
        ret = errno;
        goto done;
    }

    ret = fchown(fd, uid, gid);
    if (ret == -1) {
        ret = errno;
        goto done;
    }

    ret = set_debug_file_from_fd(fd);

done:
    free(logpath);
    return ret;
}

```

(And it's a draft, *not* tested at all)

For 1) I'd like to hear some feedback (specially) from @lslebodn as he dealt more with unprivileged users (IIRC).
For 2) I'd like to hear some feedback (specially) from @sgallagher whether he thinks it's an okay or a bad idea, as the suggestion to hack open_debug_file_ex() came from him.

@sgallagher, thanks for the review!